### ISSUE TITLE
Rubocop does not flag undeclared variables

### ISSUE THREAD

**Issue by TomasBarry** [rubocop-hq/rubocop#7226]
Given the following code:

```ruby
def x
  return if a == 1

  y = a + b
  c
end
```

Rubocop only flags a single issue:

```sh
Lint/UselessAssignment: Useless assignment to variable - 'y'
```

This can lead to subtle errors being raised when the above method is actually run.

--------

## Expected behavior

Rubocop should flag that variables 'a', 'b', and 'c' are all undeclared.

## Actual behavior

Rubocop does not flag any undeclared variables

## Steps to reproduce the problem

Using the following code:

```ruby
def x
  return if a == 1

  y = a + b
  c
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.71.0 (using Parser 2.6.3.0, running on ruby 2.6.0 x86_64-darwin18)
```

**Comment by buehmann** [comment]
This is hard because variables are not declared in Ruby. In the example above, `a`, `b`, and `c` are parsed as method calls. And those methods could be inherited and therefore defined in some other source file, making this valid and error-free code.

**Comment by TomasBarry** [comment]
You are correct. Thanks for the clarification.
